src/types/loans.ts

```diff
@@ -15,6 +15,22 @@ interface BorrowPosition extends LoanPosition {
     accumulatedDebt: MonetaryAmount<CurrencyExt>;
 }
 
+type LoanAction = "lend" | "withdraw" | "borrow" | "repay";
```

Comment:
What's the overall approach to using this `LoanAction` type? I can see that the amounts in `LoanCollateralInfo` are updated based on it, is that meant to just show quick visual feedback after a tx is sent? For instance, if a "liquidation" action occurs (or if the user withdraws/borrows etc with a script), those changes wouldn't be caught here

Reply:
This type union is useful when UI needs to compute how the LTV metrics will change once `LoanAction` is done - before submitting the tx. It is used as in  `calculateBorrowLimitBtcChange`, `calculateLtvAndThresholdsChange`.